docs/swarm.md

```diff
@@ -0,0 +1,179 @@
+format](compose-file.md#versioning) you are using:
+
+1.  If you're using version 1 along with `links`, your app will work, but Swarm
+    will schedule all containers on one host, because links between containers
+    do not work across hosts with the old networking system.
+
+2. If you're using version 2, your app should work with no changes:
+
+    - subject to the [limitations](#limitations) described below,
+
+    - as long as the Swarm cluster is configured to use the [overlay
+      driver](/engine/userguide/networking/dockernetworks.md#an-overlay-network),
+      or a custom driver which supports multi-host networking.
+
+Read the [Getting started with multi-host
+networking](/engine/userguide/networking/get-started-overlay.md) to see how to
+set up a Swarm cluster with [Docker Machine](/machine/overview) and the overlay driver.
+Once you've got it running, deploying your app to it should be as simple as:
+
+    $ eval "$(docker-machine env --swarm <name of swarm master machine>)"
+    $ docker-compose up
+
+
+## Limitations
+
+### Building images
+
+Swarm can build an image from a Dockerfile just like a single-host Docker
+instance can, but the resulting image will only live on a single node and won't
+be distributed to other nodes.
+
+If you want to use Compose to scale the service in question to multiple nodes,
+you'll have to build it yourself, push it to a registry (e.g. the Docker Hub)
+and reference it from `docker-compose.yml`:
+
+    $ docker build -t myusername/web .
+    $ docker push myusername/web
+
+    $ cat docker-compose.yml
+    web:
+      image: myusername/web
+
+    $ docker-compose up -d
+    $ docker-compose scale web=3
+
+### Multiple dependencies
+
+If a service has multiple dependencies of the type which force co-scheduling
+(see [Automatic scheduling](#automatic-scheduling) below), it's possible that
+Swarm will schedule the dependencies on different nodes, making the dependent
+service impossible to schedule. For example, here `foo` needs to be co-scheduled
+with `bar` and `baz`:
+
+    version: "2"
+    services:
+      foo:
+        image: foo
+        volumes_from: ["bar"]
+        network_mode: "service:baz"
+      bar:
+        image: bar
+      baz:
+        image: baz
+
+The problem is that Swarm might first schedule `bar` and `baz` on different
+nodes (since they're not dependent on one another), making it impossible to
+pick an appropriate node for `foo`.
+
+To work around this, use [manual scheduling](#manual-scheduling) to ensure that
+all three services end up on the same node:
+
+    version: "2"
+    services:
+      foo:
+        image: foo
+        volumes_from: ["bar"]
+        network_mode: "service:baz"
+        environment:
+          - "constraint:node==node-1"
+      bar:
+        image: bar
+        environment:
+          - "constraint:node==node-1"
+      baz:
+        image: baz
+        environment:
+          - "constraint:node==node-1"
```

Comment:
I think it makes sense to follow on from the previous (non-working) example to the same app with the workaround applied. It would be good to cover the "specific node" use case in the manual scheduling section though, so I've added an example there.